types/host.go

```diff
@@ -76,3 +81,181 @@ type HostMemoryInfo struct {
+	NrTlbLocalFlushOne         uint64 `vmstat:"nr_tlb_local_flush_one"`        // (since Linux 3.12)
+	VmacacheFindCalls          uint64 `vmstat:"vmacache_find_calls"`           // (since Linux 3.16)
+	VmacacheFindHits           uint64 `vmstat:"vmacache_find_hits"`            // (since Linux 3.16)
+	VmacacheFullFlushes        uint64 `vmstat:"vmacache_full_flushes"`         // (since Linux 3.19)
+	// the following fields are not documented in `man 5 proc` as of 4.15
+	NrZoneInactiveAnon          uint64 `vmstat:"nr_zone_inactive_anon"`
+	NrZoneActiveAnon            uint64 `vmstat:"nr_zone_active_anon"`
+	NrZoneInactiveFile          uint64 `vmstat:"nr_zone_inactive_file"`
+	NrZoneActiveFile            uint64 `vmstat:"nr_zone_active_file"`
+	NrZoneUnevictable           uint64 `vmstat:"nr_zone_unevictable"`
+	NrZoneWritePending          uint64 `vmstat:"nr_zone_write_pending"`
+	NrZspages                   uint64 `vmstat:"nr_zspages"`
+	NrShmemHugepages            uint64 `vmstat:"nr_shmem_hugepages"`
+	NrShmemPmdmapped            uint64 `vmstat:"nr_shmem_pmdmapped"`
+	AllocstallDma               uint64 `vmstat:"allocstall_dma"`
+	AllocstallDma32             uint64 `vmstat:"allocstall_dma32"`
+	AllocstallNormal            uint64 `vmstat:"allocstall_normal"`
+	AllocstallMovable           uint64 `vmstat:"allocstall_movable"`
+	PgskipDma                   uint64 `vmstat:"pgskip_dma"`
+	PgskipDma32                 uint64 `vmstat:"pgskip_dma32"`
+	PgskipNormal                uint64 `vmstat:"pgskip_normal"`
+	PgskipMovable               uint64 `vmstat:"pgskip_movable"`
+	Pglazyfree                  uint64 `vmstat:"pglazyfree"`
+	Pglazyfreed                 uint64 `vmstat:"pglazyfreed"`
+	Pgrefill                    uint64 `vmstat:"pgrefill"`
+	PgstealKswapd               uint64 `vmstat:"pgsteal_kswapd"`
+	PgstealDirect               uint64 `vmstat:"pgsteal_direct"`
+	PgscanKswapd                uint64 `vmstat:"pgscan_kswapd"`
+	PgscanDirect                uint64 `vmstat:"pgscan_direct"`
+	OomKill                     uint64 `vmstat:"oom_kill"`
+	CompactDaemonWake           uint64 `vmstat:"compact_daemon_wake"`
+	CompactDaemonMigrateScanned uint64 `vmstat:"compact_daemon_migrate_scanned"`
+	CompactDaemonFreeScanned    uint64 `vmstat:"compact_daemon_free_scanned"`
+	ThpFileAlloc                uint64 `vmstat:"thp_file_alloc"`
+	ThpFileMapped               uint64 `vmstat:"thp_file_mapped"`
+	ThpSplitPage                uint64 `vmstat:"thp_split_page"`
+	ThpSplitPageFailed          uint64 `vmstat:"thp_split_page_failed"`
+	ThpDeferredSplitPage        uint64 `vmstat:"thp_deferred_split_page"`
+	ThpSplitPmd                 uint64 `vmstat:"thp_split_pmd"`
+	ThpSplitPud                 uint64 `vmstat:"thp_split_pud"`
+	ThpSwpout                   uint64 `vmstat:"thp_swpout"`
+	ThpSwpoutFallback           uint64 `vmstat:"thp_swpout_fallback"`
+	SwapRa                      uint64 `vmstat:"swap_ra"`
+	SwapRaHit                   uint64 `vmstat:"swap_ra_hit"`
```

Comment:
Maybe this is too much info? :grimacing: Should we have an struct that can be more easily implemented for other OSs?
Even if other OSs don't have a `/proc/vmstat`, I guess that they will have some similar virtual memory stats.

Reply:
Yah, I was thinking of this. The thing is, this library makes an effort to be highly cross-platform, and we already have a lot of platform-independent higher-level memory APIs, I figured it couldn't hurt to have some more bare-metal stats. Also, I didn't want to add overhead to the main `Memory()` Call by adding extra sources to it that most users won't need. I'm not sold on it though,  since it might be a pain for clients to handle this for multiple OSes if we come up with similar low-level APIs for other OSes.

Reply:
If we did it some other way, how would the user get platform-dependent stats?